gitlab/client.py

```diff
@@ -65,6 +66,7 @@ class Gitlab:
         user_agent: A custom user agent to use for making HTTP requests.
         retry_transient_errors: Whether to retry after 500, 502, 503, 504
             or 52x responses. Defaults to False.
+        persist_base_url: reconstruct next url if found not same as user provided url
```

Comment:
I like your original wording of `keep base url`. Should we use that? It's shorter :) 

Maybe we can also a bit more specific about where it's used (might not fit in one line, just adjust accordingly if needed):

```suggestion
        keep_base_url: keep user-provided base URL for pagination if it differs from response headers
```

Reply:
done, set as suggested